charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml

```diff
@@ -292,10 +289,19 @@ spec:
 {{- if .Values.prometheus.prometheusSpec.ignoreNamespaceSelectors }}
   ignoreNamespaceSelectors: {{ .Values.prometheus.prometheusSpec.ignoreNamespaceSelectors }}
 {{- end }}
-{{- if .Values.prometheus.prometheusSpec.prometheusRulesExcludedFromEnforce }}
+{{- if .Values.prometheus.prometheusSpec.enforcedNamespaceLabel }}
```

Comment:
I moved this here and nest with the `prometheusDefaultRulesExcludedFromEnforce` and `prometheusRulesExcludedFromEnforce` as the latters are not applicable unless `enforcedNamespaceLabel` is set